WordPress/gutenberg
# Add a minHeight block support under dimensions

## Dependencies

* [x] https://github.com/WordPress/gutenberg/pull/45334

## What?


Add a `minHeight` block support feature that sits under Dimensions in the block editor inspector and in global styles.

This PR builds on the good earlier work by @aaronrobertshaw in #32499 which looked at adding in a height block support feature, and the work by @ramonjd on min-height block support in #33970.

## Why?


I think we might be at a point where adding min-height will be really useful, so let's revisit it now that we've got the style engine in 😄

Minimum height is quite useful for defining the minimum vertical dimensions of a block, while allowing it to expand beyond that size. As raised in https://github.com/WordPress/gutenberg/issues/45117 having control over minimum height could be particularly useful for blocks like Post Content when used in page layouts, and in the Group block and its Row and Stack variations.

## Issues that might be resolved or partially resolved by including this block support

* https://github.com/WordPress/gutenberg/issues/28845
* https://github.com/WordPress/gutenberg/issues/45117

## How?


* This PR mostly includes or builds on changes in the earlier preceding PRs (kudos Aaron and Ramon), but uses the style engine now.
* Add a `dimensions` key to the block supports and to the style object in block attributes, and a `minHeight` property. (I avoided the `__experimental` prefix for now, since we've been pretty settled on `dimensions` in earlier explorations, but can re-instate it if folks prefer)
* For now, use a simple `UnitControl` for the minimum height input
* Include `minHeight` in the list of controls enabled by the appearance tools opt-in

## Testing Instructions





1. In a post, add a Group block with a paragraph or two within it.
2. From the dimensions panel click to expose the `Minimum height` control and test that it works as expected.
3. In global styles, try adjusting the Group block's minimum height at the block level.
4. In a single or page template, try adjusting the minimum height of the Post Content block — it should result in being able to set a minimum height — this will likely be useful to use with `vh` units.

## Screenshots or screencast 

| Post Content block | Group block |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/197949233-ca1b65da-a0a3-4fbb-b8d8-ddc7dc16467d.png) | ![image](https://user-images.githubusercontent.com/14988353/197949274-bf9dbe41-fd87-4974-a1cb-6b4dce79fbc9.png) |
